### PR TITLE
backup/checksum: add cluster index support for ranges

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve? 

ranges only support single integer primary key.

### What is changed and how it works?

Let ranges be NotNullRange if table use the common handle

### Check List 

Tests 

 - [x] Unit test
 - [ ] Integration test
 - [x] Manual test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Related changes

 - Need to cherry-pick to the release branch

### Release note

 - No release note

### CODE REVIEW

File: pkg/backup/client.go [hunk 1]
```diff
@@ -266,8 +266,18 @@ func BuildTableRanges(tbl *model.TableInfo) ([]kv.KeyRange, error) {
 }
 
 func appendRanges(tbl *model.TableInfo, tblID int64) ([]kv.KeyRange, error) {
-	ranges := ranger.FullIntRange(false)
-	kvRanges := distsql.TableRangesToKVRanges(tblID, ranges, nil)
+	var ranges []*ranger.Range
+	if tbl.IsCommonHandle {
```
Comment (3pointer) [inline pkg/backup/client.go:270]: can we re-use the [TableHandleRangesToKVRanges](https://github.com/pingcap/tidb/blob/8fb29eb17e88cabd3968d6034d1e976823f2f79e/distsql/request_builder.go#L321) in TiDB

Comment (Little-Wallace) [reply pkg/backup/client.go:270]: Do we need to judge wether the type of pk is integer?

**Review by Little-Wallace — APPROVED** [review]
LGTM

**Pull request merged**